hls4ml/model/graph.py

```diff
@@ -865,6 +865,14 @@ class TraceData(ctypes.Structure):
         else:
             return output, trace_output
 
+    def hardware_predict(self, x, **kwargs):
```

Comment:
This method has been added to enable performing predictions directly on the FPGA from the Python code. It feels a bit intrusive to add this backend-specific code to the hls4ml core. Another approach could be to modify `predict()` to allow backend-specific overloading. So, `model.hardware_predict(x)` could become `model.predict(x, target='hw')`, but this also requires some modification of the existing core code. Could an hls4ml dev provide advice on the best approach here? (@vloncar, @jmitrevs?). Thanks!

Reply:
I would be in favor of naming this somewhat different (`predict_hw` for example) and moving the exception to backends (in `FPGABackend` should be enough to cover those that don't/cannot support it).

Some longer term idea here would be that we would have 3 ways of doing preciction: `predict_emu` (emulation, the current one), `predict_sim` (simulation via pyverilator) and `predict_hw` (real deal), with the `predict` being `predict_emu` by default with maybe a switch for user to control which one is called if it's just `predict(x)`.